php/libraries/Password.class.inc

```diff
@@ -0,0 +1,137 @@
+<?php declare(strict_types=1);
+/**
+ * This file contains a class defining a well-formed (i.e. sufficiently complex)
+ * password for LORIS.
+ *
+ * PHP Version 7
+ *
+ * @category Main
+ * @package  Main
+ * @author   John Saigle <john.saigle@mcin.ca>
+ * @license  http://www.gnu.org/licenses/gpl-3.0.txt GPLv3
+ * @link     https://www.github.com/aces/Loris
+ */
+use ZxcvbnPhp\Zxcvbn;
+
+/**
+ * Represents a well-formed password for use in the LORIS software.
+ * Automatically validates that a password is well-formed, i.e. has a sufficient
+ * length as well as complexity.
+ * Does not validate any state-related information such as whether a password
+ * exists in the database, is expired, etc. This functionality is achieved by
+ * the User class.
+ *
+ * @category Main
+ * @package  Main
+ * @author   John Saigle <john.saigle@mcin.ca>
+ * @license  http://www.gnu.org/licenses/gpl-3.0.txt GPLv3
+ * @link     https://www.github.com/aces/Loris-Trunk/
+ */
+class Password
+{
+    /**
+     * The minimum score needed to be considered a sufficiently complex password
+     * as given by the Zxcvbn password strength estimator. Passwords are
+     * evaluated by this package and scored on a scale of 0-5, with higher
+     * scores representing stronger passwords.
+     *
+     * @var  int
+     * @link https://github.com/bjeavons/zxcvbn-php
+     */
+    private const MIN_COMPLEXITY_SCORE = 3;
+
+    /**
+     * The minimum length of a valid password. This value does not need to be
+     * too long if MIN_COMPLEXITY_SCORE is high enough.
+     *
+     * @var int
+     */
+    public const MIN_PASSWORD_LENGTH = 8;
+
+    /**
+     * The result of running the $value through the password_hash algorithm.
+     *
+     * @var string
+     */
+    private $_hash;
+
+    /**
+     * Ensures that the value submitted as a potential new password is
+     * long enough.
+     *
+     * @param string $value The proposed password value.
+     *
+     * @return bool Whether the submitted value has length of at least
+     *              MIN_PASSWORD_LENGTH;
+     */
+    protected function hasSufficientLength(string $value): bool
+    {
+        // Check length
+        return strlen($value) >= self::MIN_PASSWORD_LENGTH;
+    }
+
+    /**
+     * Ensures that the value submitted as a potential new password is
+     * well-formed.
+     *
+     * @param string $value The proposed password value.
+     *
+     * @return bool Whether the submitted value meets the minimum score defined
+     *              above and given by Zxcvbn.
+     */
+    protected function hasSufficientComplexity(string $value): bool
```

Comment:
This should probably be private since it's only used inside this class.
But I wont debate school of thought if you object.